src/components/number/number.test.js

```diff
@@ -346,4 +333,169 @@ context("Tests for Number component", () => {
       }
     );
   });
+
+  describe("check accessibility for Number component", () => {
+    it.each(testData)(
+      "check accessibility when Number label using %s special characters",
+      (labelValue) => {
+        CypressMountWithProviders(<NumberInputComponent label={labelValue} />);
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(testData)(
+      "should check accessibility when labelHelp message using %s special characters",
+      (labelHelpValue) => {
+        CypressMountWithProviders(
+          <NumberInputComponent labelHelp={labelHelpValue} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(testData)(
+      "should check accessibility when fieldHelp message using %s special characters",
+      (fieldHelpValue) => {
+        CypressMountWithProviders(
+          <NumberInputComponent fieldHelp={fieldHelpValue} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(testData)(
+      "should check accessibility when prefix using %s special characters",
+      (prefixValue) => {
+        CypressMountWithProviders(
+          <NumberInputComponent prefix={prefixValue} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(testData)(
+      "should check accessibility when placeholder using %s special characters",
+      (placeholderValue) => {
+        CypressMountWithProviders(
+          <NumberInputComponent placeholder={placeholderValue} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(["11", "10"])(
+      "should check accessibility when enforce character limit of %s in Number",
+      (limit) => {
+        CypressMountWithProviders(
+          <NumberInputComponent enforceCharacterLimit characterLimit={limit} />
+        );
+
```

Comment:
we are not checking this properly - we need to type something over limit to check accessibility